vault.js

```diff
@@ -1,4 +1,28 @@
 'use strict';
-module.exports = function() {
-  
-};
\ No newline at end of file
+module.exports = function(){
+
+  const vault = {};
+
+  function setValue(key, value){
+
+    vault[key] = value;
+    return vault[key];
+  }
+
+  function getValue(key){
+
+    if (key === undefined | vault[key] === undefined){
```

Comment:
This can be simplified a little. You dont need to do two checks of key. 

There is also an edge case than should be caught.
```
var vault = VaultGenerator();

vault.getValue("toString"); // returns [Function toString]
// Should return null
```